pkg/sdkserver/sdkserver.go

```diff
@@ -435,11 +440,15 @@ func (s *SDKServer) Allocate(ctx context.Context, e *sdk.Empty) (*sdk.Empty, err
 }
 
 // Shutdown enters the Shutdown state change for this GameServer into
-// the workqueue so it can be updated
+// the workqueue so it can be updated. If gracefulTermination feature is enabled,
+// Shutdown will block on GameServer being shutdown.
 func (s *SDKServer) Shutdown(ctx context.Context, e *sdk.Empty) (*sdk.Empty, error) {
 	s.logger.Debug("Received Shutdown request, adding to queue")
 	s.stopReserveTimer()
 	s.enqueueState(agonesv1.GameServerStateShutdown)
+	if runtime.FeatureEnabled(runtime.FeatureGracefulTerminationFilter) {
```

Comment:
I'd like to see some unit tests for this please 😃

Reply:
had a stab at some tests but would like some suggestions please :)

Reply:
Tests look good! 👍🏻